### PR TITLE
Add ChiSquared distribution

### LINKED ISSUE #95: distuv: implement Chi Square distribution

### PR DESCRIPTION
**Pull request by btracey**
Fixes #95.

### CODE REVIEW

File: distuv/chisquared.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"math/rand"
+)
+
+// ChiSquared implements the ChiSquared distribution, a one parameter distribution
```
Comment (kortschak) [inline distuv/chisquared.go]: Change the second instance of "ChiSquared" to "Chi squared" or "χ²"?

File: distuv/chisquared.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+// with support on the positive numbers.
+//
+// The density function is given by
+//  1/(2^{k/2} * Γ(k/2)) * x^{k/2 - 1} * e^{-x/2}
+// It is a special case of the Gamma distribution, Γ(k/2, 1/2).
+//
+// For more information, see https://en.wikipedia.org/wiki/Chi-squared_distribution.
+type ChiSquared struct {
+	K float64 // Shape parameter, must be greater than 0.
```
Comment (kortschak) [inline distuv/chisquared.go]: Add a note that this corresponds to degrees of freedom? I anticipate that this function will see a lot of undergrads looking for ways to do tests, and not knowing what K is or how to specify dof.

File: distuv/chisquared.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+// Prob computes the value of the probability density function at x.
+func (c ChiSquared) Prob(x float64) float64 {
+	return math.Exp(c.LogProb(x))
+}
+
+// Rand returns a random sample drawn from the distribution.
+func (c ChiSquared) Rand() float64 {
+	return Gamma{c.K / 2, 0.5, nil}.Rand()
```
Comment (kortschak) [inline distuv/chisquared.go]: Nice.

**Comment by btracey** [comment]
Failure unrelated. Will fix in future PR.

File: distuv/chisquared.go (hunk 1 above)
Comment (btracey) [inline distuv/chisquared.go]: Done.

File: distuv/chisquared.go (hunk 2 above)
Comment (btracey) [inline distuv/chisquared.go]: Done.

**Pull request merged**